### PR TITLE
tabe-cache.md: add document for table cache feature

### PR DESCRIPTION
**Pull request by tiancaiamao**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: table-cache.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+```
++------+-------+
+| id   | name  |
++------+-------+
+| 1001 | Davis |
++------+-------+
+1 row in set (0.00 sec)
+```
+
+**注意： 缓存表的写入，有可能出现数秒级别的慢查询**，具体最长耗时受到全局环境变量 `@@tidb_table_cache_lease` 的控制。该变量的默认值是 3s，取值范围是 [1, 10]。
```
Comment (lcwangchao) [inline table-cache.md:158]: 可以链接到变量说明页面，并去掉 "该变量的默认值是 3s，取值范围是 [1, 10]" 防止以后调整默认值和范围的时候忘了改

File: table-cache.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+mysql> SELECT count(*) FROM t1;
++----------+
+| count(*) |
++----------+
+|   114688 |
++----------+
+1 row in set (0.04 sec)
+
+mysql> ALTER TABLE t1 CACHE;
+ERROR 8242 (HY000): 'table too large' is unsupported on cache tables.
```
Comment (lcwangchao) [inline table-cache.md:231]: 这个报错信息有点奇怪。。

File: table-cache.md (hunk 1 above)
Comment (tiancaiamao) [reply table-cache.md:158]: 这个格式怎么链接的呀，是

```
system-variables.md#tidb_txn_mode
```

还是

```
system-variables#tidb_txn_mode
```

@TomShawn

File: table-cache.md (hunk 2 above)
Comment (tiancaiamao) [reply table-cache.md:231]: 额...建议 error message 怎么改？

Comment (tiancaiamao) [reply table-cache.md:231]: 或者要不把这段代码示例给干掉？

File: table-cache.md (hunk 1 above)
Comment (TomShawn) [reply table-cache.md:158]: @tiancaiamao `/system-variables.md#tidb_txn_mode`

File: table-cache.md (hunk 2 above)
Comment (lcwangchao) [reply table-cache.md:231]: 确实干掉最方便。。。包括上面那个 select 返回的是行数其实也不能说明 "对于缓存表的大小限制为 64M" 。 所以加这段代码意义可能也不是很大

**Comment by easonn7** [comment]
LGTM

File: sql-statements/sql-statement-alter-table.md [hunk 3]
```diff
@@ -154,6 +155,7 @@ TiDB 中的 `ALTER TABLE` 语法主要存在以下限制：
 * 不支持生成列上的列类型变更。
 * 不支持部分数据类型（例如，部分时间类型、Bit、Set、Enum、JSON 等）的变更，因为 TiDB 中的 `CAST` 函数与 MySQL 的行为存在兼容性问题。
 * 不支持空间数据类型。
+* `ALTER TABLE t CACHE | NOCACHE` 不是 MySQL 标准语法，而是 TiDB 扩展功能，参见[缓存表](/cache-table.md)
```
Comment (TomShawn) [inline sql-statements/sql-statement-alter-table.md:158]:
```suggestion
* `ALTER TABLE t CACHE | NOCACHE` 不是 MySQL 标准语法，而是 TiDB 扩展功能，参见[缓存表](/cache-table.md)。
```

File: system-variables.md [hunk 4]
```diff
@@ -1652,3 +1652,11 @@ set tidb_slow_log_threshold = 200;
 - 作用域：SESSION | GLOBAL
 - 默认值：`ON`
 - 这个变量用于控制计算窗口函数时是否采用高精度模式。
+
+### `tidb_table_cache_lease`
```
Comment (TomShawn) [inline system-variables.md:1656]: 建议按做字母顺序添加在 L1523 行。
```suggestion
### `tidb_table_cache_lease` <span class="version-mark">从 v6.0.0 版本开始引入</span>
```

File: system-variables.md [hunk 5]
```diff
@@ -1652,3 +1652,11 @@ set tidb_slow_log_threshold = 200;
...
+
+- 作用域：GLOBAL
+- 默认值：`3`
+- 范围：`[1, 10]`
+- 单位：秒
+- 这个变量用来控制[缓存表](/table-cache.md)的 lease 时间，默认值是 3 秒。它会影响缓存表的修改操作，在缓存表上面执行修改，最长可能出现 `tidb_table_cache_lease` 这么久的等待。如果业务表是只读，或者可以承受很高的写入延迟，可以将这个值调大，从而增加缓存的有效时间，减少 lease 续租的频率。
```
Comment (TomShawn) [inline system-variables.md:1662]:
```suggestion
- 这个变量用来控制[缓存表](/table-cache.md)的 lease 时间，默认值是 3 秒。该变量值的大小会影响缓存表的修改。在缓存表上执行修改操作后，最长可能出现 `tidb_table_cache_lease` 变量值时长的等待。如果业务表为只读表，或者能接受很高的写入延迟，则可以将该变量值调大，从而增加缓存的有效时间，减少 lease 续租的频率。
```

File: table-cache.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
+---
+title: 缓存表
+summary: 了解 TiDB 中的缓存表功能，用于很少有修改操作的热点小表的场景，提升性能。
```
Comment (TomShawn) [inline table-cache.md:3]:
```suggestion
summary: 了解 TiDB 中的缓存表功能，用于很少被修改的热点小表，提升读性能。
```

File: table-cache.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+---
+
+# 缓存表
+
+TiDB 在 v6.0.0 版本中引入了缓存表功能。该功能针对频繁被访问，且几乎很少修改的热点小表，将表的数据直接全部加载到 TiDB 服务器的内存中，避免访问 TiKV 获取数据，从而提升性能。
```
Comment (TomShawn) [inline table-cache.md:8]:
```suggestion
TiDB 在 v6.0.0 版本中引入了缓存表功能。该功能适用于频繁被访问且很少被修改的热点小表，即把整张表的数据加载到 TiDB 服务器的内存中，直接从内存中获取表数据，避免从 TiKV 获取表数据，从而提升读性能。
```

File: table-cache.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+
+本文介绍了 TiDB 缓存表的使用场景、使用示例、与其他 TiDB 功能的兼容性限制。
+
+## 使用场景
+
+TiDB 缓存表的典型使用场景，需要满足以下条件：
```
Comment (TomShawn) [inline table-cache.md:14]:
```suggestion
TiDB 缓存表功能适用于以下特点的表：
```

File: table-cache.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+
+- 表的数据量不大
+- 只读，或者几乎很少修改
```
Comment (TomShawn) [inline table-cache.md:17]:
```suggestion
- 只读表，或者几乎很少修改
```

File: table-cache.md [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+- 表的访问很频繁，期望有更好的性能
```
Comment (TomShawn) [inline table-cache.md:18]:
```suggestion
- 表的访问很频繁，期望有更好的读性能
```

File: table-cache.md [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+
+比如配置表，业务通过这张表读取配置信息；比如金融场景中的存储汇率的表，每天只更新一次，不会实时更新；或者银行的分行或者网点信息，新增加一条记录项并不频繁。
+
+尤其是当表的数据量不大，访问又特别频繁的情况下，数据会集中在 TiKV 一个 region 上面，会形成热点，影响性能。
+以配置表为例，当业务重启的瞬间，全部连接一起加载配置，会造成较高的数据库读延迟。如果使用了缓存表，则可以解决这样的问题。
```
Comment (TomShawn) [inline table-cache.md:23]:
```suggestion
当表的数据量不大，访问又特别频繁的情况下，数据会集中在 TiKV 一个 Region 上，形成热点，从而影响性能。

因此，TiDB 缓存表的典型使用场景如下：

- 配置表，业务通过该表读取配置信息
- 金融场景中的存储汇率的表，该表不会实时更新，每天只更新一次
- 银行分行或者网点信息表，该表很少新增记录项

以配置表为例，当业务重启的瞬间，全部连接一起加载配置，会造成较高的数据库读延迟。如果使用了缓存表，则可以解决这样的问题。
```

File: table-cache.md [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+
+### 使用示例
+
```
Comment (TomShawn) [inline table-cache.md:26]:
```suggestion

#### 将普通表设为缓存表

```

File: table-cache.md [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
++-------------------------+---------+-----------+---------------+--------------------------------+
+| id                      | estRows | task      | access object | operator info                  |
++-------------------------+---------+-----------+---------------+--------------------------------+
+| UnionScan_5             | 1.00    | root      |               |                                |
+| └─TableReader_7         | 1.00    | root      |               | data:TableFullScan_6           |
+|   └─TableFullScan_6     | 1.00    | cop[tikv] | table:users   | keep order:false, stats:pseudo |
++-------------------------+---------+-----------+---------------+--------------------------------+
+3 rows in set (0.00 sec)
+```
+
```
Comment (TomShawn) [inline table-cache.md:130]:
```suggestion

#### 往缓存表写入数据

```

File: table-cache.md [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+```
+Query OK, 0 rows affected (0.01 sec)
+```
+
+
+缓存表写入延时高的原因是受到实现的限制。存在多个 TiDB 实例时，一个 TiDB 实例并不知道其它的 TiDB 实例是否缓存了数据，如果它直接执行了修改操作，而其它 TiDB 依然读取旧的缓存数据，就会读到错误的结果。
+为了保证正确性，缓存表的实现使用了一套复杂的基于 lease 的机制：读操作在缓存数据同时，还会对于缓存设置一个有效期，也就是 lease。在 lease 过期之前，保证修改操作无法执行。
+因为修改操作必须等待 lease 过期，所以会出现写入延迟。
+
+
```
Comment (TomShawn) [inline table-cache.md:177]:
```suggestion

#### 将缓存表设为普通表

```

File: table-cache.md [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE users CACHE;
+```
+
+```
+Query OK, 0 rows affected (0.01 sec)
+```
+
```
Comment (TomShawn) [inline table-cache.md:50]:
```suggestion

#### 验证是否为缓存表

```

File: table-cache.md [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+|   │ ├─executor.Compile                 | 17:47:40.533903 | 205.54µs   |
+|   │ └─session.runStmt                  | 17:47:40.534141 | 132.084µs  |
+|   │   └─TableReaderExecutor.Open       | 17:47:40.534202 | 14.749µs   |
+|   ├─*executor.UnionScanExec.Next       | 17:47:40.534306 | 3.21µs     |
+|   └─*executor.UnionScanExec.Next       | 17:47:40.534316 | 1.219µs    |
++----------------------------------------+-----------------+------------+
+7 rows in set (0.00 sec)
+```
+
+读取缓存表会用到 `UnionScan` 算子，所以通过 `explain` 看到 `UnionScan` 时，不要惊讶：
```
Comment (TomShawn) [inline table-cache.md:118]:
```suggestion
读取缓存表会使用 `UnionScan` 算子，所以通过 `explain` 查看缓存表的执行计划时，可能会在结果中看到 `UnionScan`：
```

File: table-cache.md [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+```
+
+#### 验证是否为缓存表
+
+之后观察 `SHOW CREATE TABLE` 会出现 `CACHED ON` 的属性：
+
+{{< copyable "sql" >}}
+
+```sql
+show create table users;
```
Comment (TomShawn) [inline table-cache.md:63]:
```suggestion
SHOW CREATE TABLE users;
```

File: table-cache.md [hunk 18]
```diff
@@ -0,0 +1,246 @@
...
++-------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+对缓存表的读取，会触发将数据加载到 TiDB 内存中，使用 `trace` 可以观察到，当缓存还未加载时，会访问 TiKV（出现了 `regionRequest.SendReqCtx`）。
+
+{{< copyable "sql" >}}
+
+```sql
+trace  select * from users;
```
Comment (TomShawn) [inline table-cache.md:84]: 建议将关键字大写
```suggestion
TRACE SELECT * FROM users;
```

File: table-cache.md [hunk 19]
```diff
@@ -0,0 +1,246 @@
...
+ALTER TABLE users CACHE;
+```
+
+```
+Query OK, 0 rows affected (0.01 sec)
+```
+
+#### 验证是否为缓存表
+
+之后观察 `SHOW CREATE TABLE` 会出现 `CACHED ON` 的属性：
```
Comment (TomShawn) [inline table-cache.md:58]:
```suggestion
要验证一张表是否为缓存表，使用 `SHOW CREATE TABLE` 语句。如果为缓存表，返回结果中会带有 `CACHED ON` 属性：
```

File: table-cache.md [hunk 20]
```diff
@@ -0,0 +1,246 @@
...
+| users | CREATE TABLE `users` (
+  `id` bigint(20) NOT NULL,
+  `name` varchar(100) DEFAULT NULL,
+  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin /* CACHED ON */ |
++-------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+对缓存表的读取，会触发将数据加载到 TiDB 内存中，使用 `trace` 可以观察到，当缓存还未加载时，会访问 TiKV（出现了 `regionRequest.SendReqCtx`）。
```
Comment (TomShawn) [inline table-cache.md:79]:
```suggestion
从缓存表读取数据后，TiDB 会将数据加载到内存中。你可使用 `trace` 语句查看 TiDB 是否已将数据加载到内存中。当缓存还未加载时，语句的返回结果会出现 `regionRequest.SendReqCtx`，表示 TiDB 从 TiKV 读取了数据。
```

File: table-cache.md [hunk 21]
```diff
@@ -0,0 +1,246 @@
...
+|   ├─*executor.UnionScanExec.Next               | 17:47:39.970407 | 353.073µs  |
+|   │ ├─*executor.TableReaderExecutor.Next       | 17:47:39.970411 | 301.106µs  |
+|   │ └─*executor.TableReaderExecutor.Next       | 17:47:39.970746 | 6.57µs     |
+|   └─*executor.UnionScanExec.Next               | 17:47:39.970772 | 17.589µs   |
+|     └─*executor.TableReaderExecutor.Next       | 17:47:39.970776 | 6.59µs     |
++------------------------------------------------+-----------------+------------+
+12 rows in set (0.01 sec)
+```
+
+而再次执行 `trace` 时，已经不用访问 TiKV 了：
```
Comment (TomShawn) [inline table-cache.md:107]:
```suggestion
而再次执行 `trace`，返回结果中不再有 `regionRequest.SendReqCtx`，表示 TiDB 已经不再从 TiKV 读取数据，而是直接从内存中读取：
```

File: table-cache.md [hunk 22]
```diff
@@ -0,0 +1,246 @@
...
+|   │ ├─executor.Compile                 | 17:47:40.533903 | 205.54µs   |
+|   │ └─session.runStmt                  | 17:47:40.534141 | 132.084µs  |
+|   │   └─TableReaderExecutor.Open       | 17:47:40.534202 | 14.749µs   |
+|   ├─*executor.UnionScanExec.Next       | 17:47:40.534306 | 3.21µs     |
+|   └─*executor.UnionScanExec.Next       | 17:47:40.534316 | 1.219µs    |
++----------------------------------------+-----------------+------------+
+7 rows in set (0.00 sec)
+```
+
+读取缓存表会使用 `UnionScan` 算子，所以通过 `explain` 查看缓存表的执行计划时，可能会在结果中看到 `UnionScan`：
```
Comment (TomShawn) [inline table-cache.md:124]:
```suggestion
注意，读取缓存表会使用 `UnionScan` 算子，所以通过 `explain` 查看缓存表的执行计划时，可能会在结果中看到 `UnionScan`：
```

File: table-cache.md [hunk 23]
```diff
@@ -0,0 +1,246 @@
...
+| UnionScan_5             | 1.00    | root      |               |                                |
+| └─TableReader_7         | 1.00    | root      |               | data:TableFullScan_6           |
+|   └─TableFullScan_6     | 1.00    | cop[tikv] | table:users   | keep order:false, stats:pseudo |
++-------------------------+---------+-----------+---------------+--------------------------------+
+3 rows in set (0.00 sec)
+```
+
+#### 往缓存表写入数据
+
+缓存表仍然是支持写入的，往 `users` 中插入一条记录：
```
Comment (TomShawn) [inline table-cache.md:139]:
```suggestion
缓存表支持写入数据。例如，往 `users` 表中插入一条记录：
```

File: system-variables.md [hunk 24]
```diff
@@ -1521,6 +1521,14 @@ set tidb_slow_log_threshold = 200;
 - 范围：`[0, 9223372036854775807]`
 - 这个变量用于限制 TiDB 同时向 TiKV 发送的请求的最大数量，0 表示没有限制。
 
+### `tidb_table_cache_lease`  <span class="version-mark">从 v6.0.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:1524]:
```suggestion
### `tidb_table_cache_lease` <span class="version-mark">从 v6.0.0 版本开始引入</span>
```

File: table-cache.md [hunk 25]
```diff
@@ -0,0 +1,246 @@
...
+
+```sql
+set @@global.tidb_table_cache_lease = 10;
+```
+
+```
+Query OK, 0 rows affected (0.01 sec)
+```
+
+缓存表写入延时高的原因是受到实现的限制。存在多个 TiDB 实例时，一个 TiDB 实例并不知道其它的 TiDB 实例是否缓存了数据，如果它直接执行了修改操作，而其它 TiDB 依然读取旧的缓存数据，就会读到错误的结果。为了保证正确性，缓存表的实现使用了一套复杂的基于 lease 的机制：读操作在缓存数据同时，还会对于缓存设置一个有效期，也就是 lease。在 lease 过期之前，保证修改操作无法执行。因为修改操作必须等待 lease 过期，所以会出现写入延迟。
```
Comment (TomShawn) [inline table-cache.md:178]:
```suggestion
> **注意：** 
>
> 往缓存表写入数据时，有可能出现秒级别的写入延迟。延迟的时长由全局环境变量 [`tidb_table_cache_lease`]/system-variables.md#tidb_table_cache_lease-从-v600-版本开始引入) 控制。你可根据实际业务能否承受此限制带来的延迟，决定是否适合使用缓存表功能。例如，对于完全只读的场景，可以将 `tidb_table_cache_lease` 调大：
>
> {{< copyable "sql" >}}
>
> ```sql
> set @@global.tidb_table_cache_lease = 10;
> ```
>
> 缓存表的写入延时高是受到实现的限制。存在多个 TiDB 实例时，一个 TiDB 实例并不知道其它的 TiDB 实例是否缓存了数据，如果该实例直接修改了表数据，而其它 TiDB 实例依然读取旧的缓存数据，就会读到错误的结果。为了保证数据正确性，缓存表的实现使用了一套复杂的基于 lease 的机制：读操作在缓存数据同时，还会对于缓存设置一个有效期，也就是 lease。在 lease 过期之前，无法执行对数据执行修改操作。因为修改操作必须等待 lease 过期，所以会出现写入延迟。
```

File: table-cache.md [hunk 26]
```diff
@@ -0,0 +1,246 @@
...
+
+#### 将缓存表设为普通表
+
+**注意： 对于缓存表执行 DDL 语句会失败，需要去掉缓存属性，将缓存表改回普通表后，才能执行。**
```
Comment (TomShawn) [inline table-cache.md:182]:
```suggestion
> **注意：**
>
> 对缓存表执行 DDL 语句会失败。若要对表执行 DDL语句，需要先去掉缓存属性，将缓存表设回普通表后，才能执行。
```

File: table-cache.md [hunk 27]
```diff
@@ -0,0 +1,246 @@
...
+
+```sql
+mysql> ALTER TABLE users ADD INDEX k_id(id);
+```
+
+```
+ERROR 8242 (HY000): 'Alter Table' is unsupported on cache tables.
+```
+
+通过 `ALTER TABLE t NOCACHE` 语句 可以将缓存表恢复成普通表。
```
Comment (TomShawn) [inline table-cache.md:204]:
```suggestion
使用 `ALTER TABLE t NOCACHE` 语句可以将缓存表恢复成普通表：
```

File: table-cache.md [hunk 28]
```diff
@@ -0,0 +1,246 @@
...
+set @@global.tidb_table_cache_lease = 10;
+```
+
+```
+Query OK, 0 rows affected (0.01 sec)
+```
+
+缓存表写入延时高的原因是受到实现的限制。存在多个 TiDB 实例时，一个 TiDB 实例并不知道其它的 TiDB 实例是否缓存了数据，如果它直接执行了修改操作，而其它 TiDB 依然读取旧的缓存数据，就会读到错误的结果。为了保证正确性，缓存表的实现使用了一套复杂的基于 lease 的机制：读操作在缓存数据同时，还会对于缓存设置一个有效期，也就是 lease。在 lease 过期之前，保证修改操作无法执行。因为修改操作必须等待 lease 过期，所以会出现写入延迟。
+
+#### 将缓存表设为普通表
```
Comment (TomShawn) [inline table-cache.md:180]:
```suggestion
#### 将缓存表恢复为普通表
```

File: table-cache.md [hunk 29]
```diff
@@ -0,0 +1,246 @@
...
+
+```
+Query OK, 0 rows affected (0.00 sec)
+```
+
+## 缓存表大小限制
+
+缓存表会将整张表的全部数据，加载到 TiDB 进程的内存中，并且执行修改操作后，缓存会失效，需要重新加载，因此只适用于表比较小的场景。
+
+目前 TiDB 对于缓存表的大小限制为 64M。如果表的数据超过了这个大小，执行 `ALTER TABLE t CACHE` 会失败。
```
Comment (TomShawn) [inline table-cache.md:220]:
```suggestion
由于 TiDB 将整张缓存表的数据加载到 TiDB 进程的内存中，并且执行修改操作后缓存会失效，需要重新加载，所以 TiDB 缓存表只适用于表比较小的场景。

目前 TiDB 对于每张缓存表的大小限制为 64 MB。如果表的数据超过了 64 MB，执行 `ALTER TABLE t CACHE` 会失败。
```

File: table-cache.md [hunk 30]
```diff
@@ -0,0 +1,246 @@
...
+
+## 与其他 TiDB 功能的兼容性限制
+
+以下是缓存表不支持的功能：
+
+- 不支持对分区表执行 `ALTER TABLE t CACHE` 操作
+- 不支持对临时表执行 `ALTER TABLE t CACHE` 操作
+- 不支持对视图执行 `ALTER TABLE t CACHE` 操作
+- 不支持 Stale Read 功能
+- 不支持对缓存表直接做 DDL 操作，需要先通过 `ALTER TABLE t NOCACHE` 改回普通表
```
Comment (TomShawn) [inline table-cache.md:230]:
```suggestion
- 不支持对缓存表直接做 DDL 操作，需要先通过 `ALTER TABLE t NOCACHE` 将缓存表改回普通表后再进行 DDL 操作。
```

File: table-cache.md [hunk 31]
```diff
@@ -0,0 +1,246 @@
...
+
+以下是缓存表无法使用缓存的场景：
+
+- 设置系统变量 `tidb_snapshot` 读取历史数据
+- 执行修改操作期间，会使缓存失效，直到下次数据被再次加载
```
Comment (TomShawn) [inline table-cache.md:235]:
```suggestion
- 执行修改操作期间，已有缓存会失效，直到数据被再次加载
```

File: table-cache.md [hunk 32]
```diff
@@ -0,0 +1,246 @@
...
+
+## TiDB 生态工具兼容性
+
+缓存表并不是标准 MySQL 功能，而是 TiDB 扩展，并且只有 TiDB 识别 `ALTER TABLE CACHE` 语句。所有的 TiDB 生态工具，并不支持这一项功能，包括 br, cdc, dumpling 等组件，它们会将缓存表当作普通表处理。
+
+这意味着，备份恢复一张缓存表，它会变成一张普通表。如果下游集群是另一套 TiDB 并且您希望仍然用上缓存表，可以对下游集群中的表执行 `ALTER TABLE CACHE` 手动开启。
```
Comment (TomShawn) [inline table-cache.md:241]:
```suggestion
缓存表并不是标准的 MySQL 功能，而是 TiDB 扩展。只有 TiDB 能识别 `ALTER TABLE CACHE` 语句。所有的 TiDB 生态工具均不支持缓存表功能，包括 Backup & Restore (BR)、TiCDC、Dumpling 等组件，它们会将缓存表当作普通表处理。

这意味着，备份恢复一张缓存表时，它会变成一张普通表。如果下游集群是另一套 TiDB 集群并且你希望继续使用缓存表功能，可以对下游集群中的表执行 `ALTER TABLE CACHE` 手动开启缓存表功能。
```

File: system-variables.md [hunk 33]
```diff
@@ -1521,6 +1521,14 @@ set tidb_slow_log_threshold = 200;
 - 范围：`[0, 9223372036854775807]`
 - 这个变量用于限制 TiDB 同时向 TiKV 发送的请求的最大数量，0 表示没有限制。
 
+### `tidb_table_cache_lease` <span class="version-mark">从 v6.0.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:1524]:
```suggestion
### `tidb_table_cache_lease`（从 v6.0.0 版本开始引入）
```

File: table-cache.md [hunk 34]
```diff
@@ -0,0 +1,246 @@
...
+| id   | name  |
++------+-------+
+| 1001 | Davis |
++------+-------+
+1 row in set (0.00 sec)
+```
+
+> **注意：** 
+>
+> 往缓存表写入数据时，有可能出现秒级别的写入延迟。延迟的时长由全局环境变量 [`tidb_table_cache_lease`]/system-variables.md#tidb_table_cache_lease-从-v600-版本开始引入) 控制。你可根据实际业务能否承受此限制带来的延迟，决定是否适合使用缓存表功能。例如，对于完全只读的场景，可以将 `tidb_table_cache_lease` 调大：
```
Comment (TomShawn) [inline table-cache.md:168]:
```suggestion
> 往缓存表写入数据时，有可能出现秒级别的写入延迟。延迟的时长由全局环境变量 [`tidb_table_cache_lease`](/system-variables.md#tidb_table_cache_lease从-v600-版本开始引入) 控制。你可根据实际业务能否承受此限制带来的延迟，决定是否适合使用缓存表功能。例如，对于完全只读的场景，可以将 `tidb_table_cache_lease` 调大：
```

File: table-cache.md [hunk 35]
```diff
@@ -0,0 +1,246 @@
...
+当表的数据量不大，访问又特别频繁的情况下，数据会集中在 TiKV 一个 Region 上，形成热点，从而影响性能。因此，TiDB 缓存表的典型使用场景如下：
+
+- 配置表，业务通过该表读取配置信息
+- 金融场景中的存储汇率的表，该表不会实时更新，每天只更新一次
+- 银行分行或者网点信息表，该表很少新增记录项
+
+以配置表为例，当业务重启的瞬间，全部连接一起加载配置，会造成较高的数据库读延迟。如果使用了缓存表，则可以解决这样的问题。
+
+### 使用示例
+
```
Comment (TomShawn) [inline table-cache.md:29]:
```suggestion

本节通过示例介绍缓存表的使用方法。

```

File: table-cache.md [hunk 36]
```diff
@@ -0,0 +1,246 @@
...
+>
+> 往缓存表写入数据时，有可能出现秒级别的写入延迟。延迟的时长由全局环境变量 [`tidb_table_cache_lease`]/system-variables.md#tidb_table_cache_lease-从-v600-版本开始引入) 控制。你可根据实际业务能否承受此限制带来的延迟，决定是否适合使用缓存表功能。例如，对于完全只读的场景，可以将 `tidb_table_cache_lease` 调大：
+>
+> {{< copyable "sql" >}}
+>
+> ```sql
+> set @@global.tidb_table_cache_lease = 10;
+> ```
+>
+> 缓存表的写入延时高是受到实现的限制。存在多个 TiDB 实例时，一个 TiDB 实例并不知道其它的 TiDB 实例是否缓存了数据，如果该实例直接修改了表数据，而其它 TiDB 实例依然读取旧的缓存数据，就会读到错误的结果。为了保证数据正确性，缓存表的实现使用了一套复杂的基于 lease 的机制：读操作在缓存数据同时，还会对于缓存设置一个有效期，也就是 lease。在 lease 过期之前，无法执行对数据执行修改操作。因为修改操作必须等待 lease 过期，所以会出现写入延迟。
```
Comment (TomShawn) [inline table-cache.md:176]:
```suggestion
> 缓存表的写入延时高是受到实现的限制。存在多个 TiDB 实例时，一个 TiDB 实例并不知道其它的 TiDB 实例是否缓存了数据，如果该实例直接修改了表数据，而其它 TiDB 实例依然读取旧的缓存数据，就会读到错误的结果。为了保证数据正确性，缓存表的实现使用了一套基于 lease 的复杂机制：读操作在缓存数据同时，还会对于缓存设置一个有效期，也就是 lease。在 lease 过期之前，无法对数据执行修改操作。因为修改操作必须等待 lease 过期，所以会出现写入延迟。
```

File: table-cache.md [hunk 37]
```diff
@@ -0,0 +1,246 @@
...
+
+#### 将缓存表恢复为普通表
+
+> **注意：**
+>
+> 对缓存表执行 DDL 语句会失败。若要对表执行 DDL语句，需要先去掉缓存属性，将缓存表设回普通表后，才能执行。
```
Comment (TomShawn) [inline table-cache.md:182]:
```suggestion
> 对缓存表执行 DDL 语句会失败。若要对缓存表执行 DDL 语句，需要先去掉缓存属性，将缓存表设回普通表后，才能执行 DDL 语句。
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

File: table-cache.md (hunk 34 above)
Comment (tiancaiamao) [reply table-cache.md:168]: 这个链接写法对么？或是应该这样子？
[`tidb_table_cache_lease`](/system-variables.md#tidb_table_cache_lease)

Comment (TomShawn) [reply table-cache.md:170]: 标题后面有 `（从 v6.0.0 版本开始引入）` 这一块，所以锚点也要加上

**Review by lcwangchao — APPROVED** [review]

**Comment by TomShawn** [comment]
follow-up: add TOC

**Comment by TomShawn** [comment]
@ran-huang 新增术语”缓存表/Cached Table“：

zh：TiDB 把整张表的数据加载到服务器的内存中，直接从内存中获取表数据，避免从 TiKV 获取表数据，从而提升读性能。

en： TiDB loads the data of an entire table into the memory of the TiDB server, and TiDB directly gets the table data from the memory without accessing TiKV, which improves the read performance.

**Pull request merged**